monkey/agent_plugins/util.sh

```diff
@@ -33,15 +34,23 @@ lower() {
 get_plugin_filename() {
     _plugin_path=${1:-"."}
 
-    # if "manifest.yaml" doesn't exist, assume the file is called "manifest.yml"
-    # the script will fail if that doesn't exist either
-    if [ ! -f "${_plugin_path}/$MANIFEST_FILENAME" ]; then
-        MANIFEST_FILENAME=manifest.yml
-    fi
+    MANIFEST_FILENAME=$(get_plugin_manifest_filename)
 
     _name=$(get_value_from_key "${_plugin_path}/$MANIFEST_FILENAME" name) || fail "Failed to get plugin name"
     _name=$(ltrim "$_name")
     _type=$(get_value_from_key "${_plugin_path}/$MANIFEST_FILENAME" plugin_type) || fail "Failed to get plugin type"
     _type=$(ltrim "$(lower "$_type")")
     echo "${_name}-${_type}.tar"
 }
+
+get_plugin_manifest_filename() {
+    _plugin_path=${1:-"."}
+
+    # if "manifest.yaml" doesn't exist, assume the file is called "manifest.yml"
+    # the script will fail if that doesn't exist either
```

Comment:
Nothing in this function will cause the script to fail if neither file exists, so this comment is a bit confusing. It seems like the comment is wrong. I'm assuming the comment is referring to some behavior outside of this function, but that's dangerous. if that behavior changes, this comment will be wrong, and a person would be unlikely to fix this comment if they changed that behavior. Also a bit spaghetti-ish.